packages/postgres-database/tests/test_users_details.py

```diff
@@ -3,135 +3,517 @@
+):
+    """Test that billing details can be retrieved from pre-registration data."""
+    pre_email, fake_pre_registration_data = pre_registered_user
+
+    # Create the user
+    async with transaction_context(asyncpg_engine) as connection:
+        new_user = await UsersRepo.new_user(
+            connection,
+            email=pre_email,
+            password_hash="123456",  # noqa: S106
+            status=UserStatus.ACTIVE,
+            expires_at=None,
+        )
+        await UsersRepo.link_and_update_user_from_pre_registration(
+            connection, new_user_id=new_user.id, new_user_email=new_user.email
+        )
+
+    # Get billing details
+    async with pass_or_acquire_connection(asyncpg_engine) as connection:
+        invoice_data = await UsersRepo.get_billing_details(
+            connection, user_id=new_user.id
+        )
+        assert invoice_data is not None
+
+    # Test UserAddress model conversion
+    user_address = UserAddress.create_from_db(invoice_data)
 
+    # Verify address fields match the pre-registration data
     assert user_address.line1
     assert user_address.state == fake_pre_registration_data["state"]
     assert user_address.postal_code == fake_pre_registration_data["postal_code"]
     assert user_address.country == fake_pre_registration_data["country"]
 
-    # now let's update the user
-    result = await connection.execute(
-        users.update()
-        .values(first_name="My New Name")
-        .where(users.c.id == new_user.id)
-        .returning("*")
-    )
-    updated_user = await result.fetchone()
+
+@pytest.mark.acceptance_test(
+    "pre-registration user update in https://github.com/ITISFoundation/osparc-simcore/issues/5138"
+)
+async def test_update_user_from_pre_registration(
+    asyncpg_engine: AsyncEngine,
+    pre_registered_user: tuple[str, dict[str, Any]],
+):
+    """Test that pre-registration details override manual updates when re-linking."""
+    pre_email, _ = pre_registered_user
+
+    # Create the user and link to pre-registration
+    async with transaction_context(asyncpg_engine) as connection:
+        new_user = await UsersRepo.new_user(
+            connection,
+            email=pre_email,
+            password_hash="123456",  # noqa: S106
+            status=UserStatus.ACTIVE,
+            expires_at=None,
+        )
+        await UsersRepo.link_and_update_user_from_pre_registration(
+            connection, new_user_id=new_user.id, new_user_email=new_user.email
+        )
+
+    # Update the user manually
+    async with transaction_context(asyncpg_engine) as connection:
+        result = await connection.execute(
+            users.update()
+            .values(first_name="My New Name")
+            .where(users.c.id == new_user.id)
+            .returning("*")
+        )
+        updated_user = result.one()
 
     assert updated_user
     assert updated_user.first_name == "My New Name"
     assert updated_user.id == new_user.id
 
-    for _ in range(2):
```

Comment:
I think this should still be part of the test. Looks to me that it was done to make sure that the actions are repetable and don't raise errors.
This is now missing from the refactored version